src/OnlineSales/Controllers/ImagesController.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.Net.Http.Headers;
+using OnlineSales.Data;
+using OnlineSales.DTOs;
+using OnlineSales.Entities;
+using Quartz.Util;
+
+namespace OnlineSales.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ImagesController : ControllerBase
+    {
+        private readonly ApiDbContext apiDbContext;
+
+        public ImagesController(ApiDbContext apiDbContext)
+        {
+            this.apiDbContext = apiDbContext;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult> Post([FromForm] ImageDtos imageDtos)
+        {
+            var provider = new FileExtensionContentTypeProvider();
+
+            string incomingFileName = imageDtos.Image!.FileName;
+            string incomingFileExtension = Path.GetExtension(imageDtos.Image!.FileName);
+            long incomingFileSize = imageDtos.Image!.Length; // bytes
+            string? incomingFileMimeType = string.Empty;
+            string returnedFileName = RandomString(5) + DateTime.Now.Year.ToString() + new Random().Next(1000, 9999).ToString() + "-" + DateTime.Now.Month.ToString("00") + DateTime.Now.Day.ToString("00") + RandomString(5);
+
+            provider.TryGetContentType(incomingFileName, out incomingFileMimeType);
+
+            using var fileStream = imageDtos.Image.OpenReadStream();
+            byte[] imageInBytes = new byte[incomingFileSize];
+            fileStream.Read(imageInBytes, 0, (int)imageDtos.Image.Length);
+
+            string baseUrl = $"{Request.Scheme}://{Request.Host.Value}{Request.Path}";
+
+            UploadedImage uploadedImage = new ()
+            {
+                FileName = incomingFileName,
+                FileSize = incomingFileSize,
+                ImageBinaryData = imageInBytes,
+                MimeType = incomingFileMimeType!,
+                ScopeId = imageDtos.ScopeId,
+                ReturnedFileName = returnedFileName,
+                FileExtension = incomingFileExtension,
+                CreatedAt = DateTime.UtcNow,
+                CreatedByIP = GetClientIP(),
+                CreatedByUserAgent = GetUserAgent(),
+            };
+
+            await apiDbContext.UploadedImages!.AddAsync(uploadedImage);
+
+            await apiDbContext.SaveChangesAsync();
+
+            return Ok(new { ImageUrl = baseUrl + "/" + imageDtos.ScopeId + "/" + returnedFileName + incomingFileExtension });
+        }
+
+        [Route("{scopeId}/{fileName}")]
+        [ResponseCache(VaryByHeader = "User-Agent", Duration = 120)]
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult> Get(string scopeId, string fileName)
+        {
+            if (scopeId.IsNullOrWhiteSpace())
+            {
+                return BadRequest("Scope is invalid");
```

Comment:
@WASandaruwan you return BadRequest which is HTTP 400 status code, but you have not declared this status code as potential ProducesResponseType for this method. Please do not just copy and paste what you see in other methods, think why is it done this way and how does it correlate with the code you are writing... You code here will not produce 100% correct swagger definition.

Reply:
Added [ProducesResponseType(StatusCodes.Status400BadRequest)]